backend/googlecloudstorage/googlecloudstorage.go

```diff
@@ -798,6 +845,26 @@ func (f *Fs) makeBucket(ctx context.Context, bucket string) (err error) {
 // Returns an error if it isn't empty: Error 409: The bucket you tried
 // to delete was not empty.
 func (f *Fs) Rmdir(ctx context.Context, dir string) (err error) {
+	// Add trailing slash to folder path
+	dirFilePath := fmt.Sprintf("%s/", dir)
+	if len(f.rootDirectory) != 0 {
+		dirFilePath = fmt.Sprintf("%s/%s", f.rootDirectory, dirFilePath)
+	}
+	// Query directory marker object before attempting to remove it.
+	e := f.pacer.Call(func() (bool, error) {
+		_, e := f.svc.Objects.Get(f.rootBucket, dirFilePath).Context(ctx).Do()
+		return shouldRetry(ctx, e)
+	})
+	if e == nil {
+		// Using API directly due to trailing slash being removed before Remove() can get full path
```

Comment:
I'd prefer to call Object.Remove here or something factored out of Object.Remove rather than duplicate this.

Reply:
Both Upload and Remove are expanding bucket and bucketPath from o.split() that looses the directory marker filename (`/`) after it's run trough `path.Join(f.root, rootRelativePath)`. Basically, it trims the trailing slash that would be the "filename" of that marker.
See this playground example: https://play.golang.org/p/m8Y1OyvKw1f
Due to this, custom functions are used. I could try replacing `bucket, bucketPath := o.split()`, runing tests and checking if they still pass. The other option is still keeping the duplicate functions (that I'd love to avoid too).

Reply:
This is the best i could do for split() 
```
// path.Join might stripped trailing slash that can be used as an object name for bucket based remotes.
if strings.HasSuffix(rootRelativePath, "/") {
	bucketPath += "/"
}
```

Reply:
The snippet above did it, This works now.